### ISSUE TITLE
The TypeScript language service died unexpectedly 5 times in the last 5 Minutes.

### ISSUE THREAD

**Issue by holandaleo** [microsoft/vscode#88163]
Issue Type: <b>Bug</b>

Potential duplicate of [#20446](https://github.com/microsoft/TypeScript/issues/20446). I could have created the issue under TypeScript project, but the "Report on Github" button lead me here, and TypeScript service did not show as an option in the bug report form. Feel free to contact me about more information on my settings.

This happens all the time regardless of what I am doing. Crash notification keeps showing up over and over again.

VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:57:51.166Z)
OS version: Darwin x64 19.2.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (4 x 2300)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|6, 5, 4|
|Memory (System)|8.00GB (0.50GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details><details><summary>Extensions (8)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|2.0.11
gitlens|eam|10.2.0
mssql|ms-|1.8.0
vsliveshare|ms-|1.0.1404
trailing-spaces|sha|0.3.1
js-jsx-snippets|sky|10.0.0
JavaScriptSnippets|xab|1.7.2
ReactSnippets|xab|2.4.0

</details>

**Comment by holandaleo** [comment]
I just found more details on that... I went to *Help > Toggle Developer Tools"  and noticed that every few seconds this error message goes to error console:

```
Error: <semantic> TypeScript Server Error (3.7.3)
Debug Failure. False expression.
Error: Debug Failure. False expression.
    at declareSymbol (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29212)
    at declareSymbolAndAddToSymbolTable (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30497)
    at bindThisPropertyAssignment (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:31369)
    at bindWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:31020)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30902)
    at bindExpressionStatement (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30106)
    at bindChildrenWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29568)
    at bindChildren (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29489)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30913)
    at /Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29495
    at bindEach (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29511)
    at bindEachFunctionsFirst (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29495)
    at bindChildrenWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29611)
    at bindChildren (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29489)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30913)
    at visitNode (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17622)
    at Object.forEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17752)
    at bindEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29519)
    at bindChildrenWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29614)
    at bindChildren (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29489)
    at bindContainer (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29435)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30916)
    at visitNode (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17622)
    at Object.forEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17709)
    at bindEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29519)
    at bindChildrenWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29614)
    at bindChildren (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29489)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30913)
    at bindEach (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29511)
    at visitNodes (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17627)
    at Object.forEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17801)
    at bindEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29519)
    at bindChildrenWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29614)
    at bindChildren (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29489)
    at bindContainer (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29470)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30916)
    at visitNode (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17622)
    at Object.forEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:17996)
    at bindEachChild (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29519)
    at bindChildrenWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29614)
    at bindChildren (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29489)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30913)
    at /Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29495
    at bindEach (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29511)
    at bindEachFunctionsFirst (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29495)
    at bindChildrenWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29605)
    at bindChildren (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29489)
    at bindContainer (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29435)
    at bind (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:30916)
    at bindSourceFile (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29068)
    at Object.bindSourceFile (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:29005)
    at initializeTypeChecker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:64384)
    at Object.createTypeChecker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:33622)
    at Object.getTypeChecker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:95612)
    at synchronizeHostData (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:130645)
    at Object.getProgram (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:130727)
    at ConfiguredProject.Project.updateGraphWorker (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:135078)
    at ConfiguredProject.Project.updateGraph (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:135021)
    at ConfiguredProject.updateGraph (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:135869)
    at ProjectService.createLoadAndUpdateConfiguredProject (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:137362)
    at ProjectService.assignProjectToOpenedScriptInfo (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:138186)
    at /Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:138417
    at Object.mapDefined (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:565)
    at ProjectService.applyChangesInOpenFiles (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:138417)
    at Session.handlers.ts.createMapFromTemplate._a.<computed> (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:139271)
    at /Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:140929
    at IOSession.Session.executeWithRequestId (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:140920)
    at IOSession.Session.executeCommand (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:140929)
    at IOSession.Session.onMessage (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:140952)
    at Interface.<anonymous> (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/tsserver.js:142267)
    at Interface.emit (events.js:200)
    at Interface._onLine (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/readline.js:314)
    at Interface._normalWrite (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/readline.js:459)
    at Socket.ondata (/Applications/Visual%20Studio%20Code.app/Contents/Resources/app/out/vs/code/electron-browser/workbench/readline.js:170)
    at Socket.emit (events.js:200)
    at addChunk (_stream_readable.js:294)
    at readableAddChunk (_stream_readable.js:275)
    at Socket.Readable.push (_stream_readable.js:210)
    at Pipe.onStreamRead (internal/stream_base_commons.js:166)
	at Function.create (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:1)
	at h.dispatchResponse (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:1)
	at h.dispatchMessage (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:1)
	at /Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:1
	at l.fire (/Applications/Visual Studio Code.app/Contents/Resources/app/out/vs/workbench/services/extensions/node/extensionHostProcess.js:46)
	at t.Reader.onLengthData (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:1)
	at Socket.<anonymous> (/Applications/Visual Studio Code.app/Contents/Resources/app/extensions/typescript-language-features/dist/extension.js:1)
	at Socket.emit (events.js:200)
	at addChunk (_stream_readable.js:294)
	at readableAddChunk (_stream_readable.js:275)
	at Socket.Readable.push (_stream_readable.js:210)
	at Pipe.onStreamRead (internal/stream_base_commons.js:166)
```

**Comment by holandaleo** [comment]
Also tried with no extensions running. Same issue.

Issue Type: <b>Bug</b>

Updated

VS Code version: Code 1.41.1 (26076a4de974ead31f97692a0d32f90d735645c0, 2019-12-18T14:57:51.166Z)
OS version: Darwin x64 19.2.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7360U CPU @ 2.30GHz (4 x 2300)|
|GPU Status|2d_canvas: enabled<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>metal: disabled_off<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>protected_video_decode: unavailable_off<br>rasterization: enabled<br>skia_renderer: disabled_off<br>surface_control: disabled_off<br>surface_synchronization: enabled_on<br>video_decode: enabled<br>viz_display_compositor: enabled_on<br>viz_hit_test_surface_layer: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|7, 6, 5|
|Memory (System)|8.00GB (0.13GB free)|
|Process Argv||
|Screen Reader|no|
|VM|0%|
</details>Extensions: none

**Comment by mjbvz** [comment]
Please try upgrading your workspace to use `typescript@next` by installing [this extension](https://marketplace.visualstudio.com/items?itemName=ms-vscode.vscode-typescript-next). Does that fix the issue?

**Comment by holandaleo** [comment]
Yea! That solved the problem!

**Comment by mjbvz** [comment]
Closing as fixed in TS 3.8. We will be picking that version of TS up for the February release of VS Code. Until then, use the extension
